src/lustre/lustreAstNormalizer.ml

```diff
@@ -711,7 +722,40 @@ and normalize_item info map = function
   | FrameBlock _ -> 
     assert false
   | AnnotMain (pos, b) -> AnnotMain (pos, b), empty (), []
-  | AnnotProperty (pos, name, expr) ->
+  | AnnotProperty (pos, name, expr, k) -> 
+    
+    let expr = (match k with 
+      (* expr or counter < b *)
+      | Reachable Some (From b) -> 
+        A.BinaryOp (pos, Or, A.UnaryOp (pos, A.Not, expr), 
+        A.CompOp(pos, A.Lt, Ident(dpos, ctr_id), 
+                            Const (dpos, Num (HString.mk_hstring (string_of_int b)))))
+
+      (* expr or counter != b *)
+      | Reachable Some (At b) -> 
+        A.BinaryOp (pos, Or, A.UnaryOp (pos, A.Not, expr), 
+        A.CompOp(pos, A.Neq, Ident(dpos, ctr_id), 
+                             Const (dpos, Num (HString.mk_hstring (string_of_int b)))))
+
+      (* expr or counter > b *)
+      | Reachable Some (Within b) -> 
+        A.BinaryOp (pos, Or, A.UnaryOp (pos, A.Not, expr), 
+        A.CompOp(pos, A.Gt, Ident(dpos, ctr_id), 
+                            Const (dpos, Num (HString.mk_hstring (string_of_int b)))))
+      
+      (* expr or counter < b1 or counter > b2 *)
+      | Reachable Some (FromWithin (b1, b2)) -> 
+        A.BinaryOp (pos, Or, A.UnaryOp (pos, A.Not, expr), 
+        A.BinaryOp (pos, Or, 
+          A.CompOp(pos, A.Lt, Ident(dpos, ctr_id), 
+          Const (dpos, Num (HString.mk_hstring (string_of_int b1)))),
+          A.CompOp(pos, A.Gt, Ident(dpos, ctr_id), 
+                              Const (dpos, Num (HString.mk_hstring (string_of_int b2)))))
+        )
+
+      | Reachable _ -> A.UnaryOp (pos, A.Not, expr)
+      | _ -> expr
+    ) in
```

Comment:
If you swap the order of this `let expr` binding and the `let name'` binding below, then reachability properties without a name will use the original expression instead of the negated expression and calls to the `rm_neg_prop_str` function in `kEvent` will not be necessary anymore.